funfact/algorithm.py

```diff
@@ -1,18 +1,22 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 import numpy as np
-import tqdm
 import funfact.optim
 import funfact.loss
 from funfact import Factorization
 from funfact.backend import active_backend as ab
 from funfact.vectorization import view
+from funfact.gradient_descent import (
+    gradient_descent,
+    gradient_descent_plugin,
+    GradientDescentState,
+)
 
 
 def factorize(
     tsrex, target, optimizer='Adam', loss='MSE', lr=0.1, tol=1e-6,
     max_steps=10000, vec_size=1, vec_axis=0, stop_by='first', returns='best',
-    checkpoint_freq=50, dtype=None, penalty_weight=1.0
+    checkpoint_freq=50, dtype=None, penalty_weight=1.0, plugins=[]
```

Comment:
Should we add something about the `plugin` argument to the doc string below?

Reply:
Good point. In fact, the entire plugin system should be documented, probably via a later PR. For this one I'll add a simple one-liner for the time being.